python/cudf/cudf/_lib/pylibcudf/strings/regex_program.pyx

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+
+from libcpp.memory cimport unique_ptr
+from libcpp.string cimport string
+from libcpp.utility cimport move
+
+from cudf._lib.pylibcudf.libcudf.strings.regex_flags cimport regex_flags
+from cudf._lib.pylibcudf.libcudf.strings.regex_program cimport regex_program
+
+from cudf._lib.pylibcudf.strings.regex_flags import RegexFlags
+
+
+cdef class RegexProgram:
+
+    def __init__(self, *args, **kwargs):
+        raise ValueError("Do not instantiate RegexProgram directly, use create")
+
+    @staticmethod
+    def create(object pattern, object flags):
+        cdef unique_ptr[regex_program] c_prog
+        cdef regex_flags c_flags
+        cdef string c_pattern = pattern.encode()
+
+        cdef RegexProgram ret = RegexProgram.__new__(RegexProgram)
+        if isinstance(flags, object):
+            if isinstance(flags, (int, RegexFlags)):
+                c_flags = flags
+                with nogil:
+                    c_prog = regex_program.create(c_pattern, c_flags)
```

Comment:
*question*: Whose job is it to ensure that the provided input is a "valid" regex?

Reply:
The libcudf [method](https://docs.rapids.ai/api/libcudf/legacy/structcudf_1_1strings_1_1regex__program#a3100141866b6b2858d014e36287e5d0a) should validate the pattern at the c++ level:

```
cudf::logic_error If pattern is invalid or contains unsupported features
```

Reply:
Can we add some simple tests of this?

Reply:
Yeah I think that's reasonable. Will add some

Reply:
Couple of tests at https://github.com/rapidsai/cudf/pull/15880/commits/b15588a484cd2151cbf65d0d32cd9bec870baf4b